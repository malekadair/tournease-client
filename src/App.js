import React from "react";
import "./App.css";

function App() {
  return <div>√ç</div>;
}

export default App;
