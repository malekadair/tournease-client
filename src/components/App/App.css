html,
body,
div,
span,
applet,
object,
iframe,
h1,
h2,
h3,
h4,
h5,
h6,
p,
blockquote,
pre,
a,
abbr,
acronym,
address,
big,
cite,
code,
del,
dfn,
em,
img,
ins,
kbd,
q,
s,
samp,
small,
strike,
strong,
sub,
sup,
tt,
var,
b,
u,
i,
center,
dl,
dt,
dd,
ol,
ul,
li,
fieldset,
form,
label,
legend,
table,
caption,
tbody,
tfoot,
thead,
tr,
th,
td,
article,
aside,
canvas,
details,
embed,
figure,
figcaption,
footer,
header,
hgroup,
menu,
nav,
output,
ruby,
section,
summary,
time,
mark,
audio,
video {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}
/* HTML5 display-role reset for older browsers */
article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
  display: block;
}
body {
  line-height: 1;
}
ol,
ul {
  list-style: none;
}
blockquote,
q {
  quotes: none;
}
blockquote:before,
blockquote:after,
q:before,
q:after {
  content: "";
  content: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}

/* CSS RESET ABOVE THIS */

nav {
  display: flex;
  justify-content: space-between;
  min-height: 10vh;
  background: rgb(196, 166, 166);
  flex-direction: column;
  align-items: center;
  padding: 5px;
}

.nav-links {
  display: flex;
  width: 300px;
  justify-content: space-evenly;
  align-items: center;
  list-style: none;
  font-size: 25px;
}

@import url("https://fonts.googleapis.com/css?family=Lato:200, 400");
@import url("https://fonts.googleapis.com/css?family=Lustria");

* {
  box-sizing: border-box;
}
html {
  background-color: rgb(210, 210, 210);
}
.mainContent {
  text-align: center;
}
.login {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 30px;
  margin: auto;
}

.home {
  display: flex;
  padding: 30px;
  flex-direction: column;
  align-items: center;
}

#upcoming {
  text-align: center;
}

.aButton,
button {
  color: #571717 !important;
  /* text-transform: uppercase; */
  background: #ffffff;
  font-size: 15px;
  font-weight: 500;
  padding: 5px 10px;
  margin: 5px;
  border: 3px solid #555 !important;
  border-radius: 6px;
  display: inline-block;
  transition: all 0.3s ease 0s;
}

.aButton:hover,
button:hover {
  color: #494949 !important;
  border-radius: 15px;
  border-color: #571717 !important;
  transition: all 0.3s ease 0s;
  cursor: pointer;
}
.tournamentList {
  display: flex;
  justify-content: space-evenly;
  flex-wrap: wrap;
  align-items: center;
}

.tournament {
  /* display: flex;
   */
  text-align: center;
  background-color: white;
  width: 350px;
  height: 200px;
  border-radius: 10px;
  padding: 10px;
  margin: 15px 10px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.tournament:hover {
  background-color: rgb(252, 240, 240);
}
.cardText {
  font-size: 20px;
  margin: 2px 0;
}
.details {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.tourneyTitle {
  margin: 20px;
  font-size: 30px;
  font-weight: 800;
}
.tourneyText {
  margin: 5px;
}
body {
  margin: 0;
  padding: 0;
  font-family: "Lato", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.create {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.loginText {
  margin-top: 10px;
}

.formItem {
  display: flex;
  flex-wrap: wrap;
  font-size: 20px;
  align-items: center;
}
input,
select,
textarea {
  border-radius: 4px;
  border: 2px solid #555;
  margin: 0 5px;
  width: 175px;
}
input,
select {
  height: 25px;
}
form {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 40px;
  max-width: 600px;
  width: 100%;
}
h1 {
  font-size: 50px;
}

h2 {
  font-size: 25px;
}

a {
  text-decoration: none;
  color: #571717;
}

a:hover {
  color: rgb(0, 0, 0);
  /* text-decoration: underline; */
}

h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: "Lustria";
  font-weight: 100;
  color: #222;
}

p {
  color: #555;
}

.blue {
  color: #477acc;
}

.yellow {
  color: #eac435;
}

.green {
  color: #09a55e;
}

.red {
  color: #ca1551;
}

.orange {
  color: #fb4d3d;
}
