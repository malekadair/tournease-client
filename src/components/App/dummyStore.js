export default {
  tourneys: [
    {
      id: 1,
      title: "US Open",
      date: "11/1/1991",
      time: "2pm",
      game: "9-Ball",
      fee: "",
      address: "",
      moreDetails: ""
    }
  ]
};
