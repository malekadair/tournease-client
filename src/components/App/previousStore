{
      id: 1,
      name: "happy",
      game: "8-ball",
      details:
        "Deserunt tempor deserunt commodo et ad eu minim nostrud culpa nisi exercitation sint. Officia minim enim ipsum est veniam Lorem amet labore fugiat sint amet laboris. Mollit aliqua commodo minim id sunt cillum est velit do minim proident nulla mollit reprehenderit. Elit proident cillum culpa id dolore. Incididunt commodo commodo excepteur labore aliqua ea ipsum ipsum aute. Laboris ipsum ullamco velit proident non est nostrud esse nulla fugiat."
    },
    {
      id: 2,
      name: "sad",
      game: "9-ball",
      details:
        "Enim elit eu id Lorem irure pariatur consectetur cillum tempor dolore proident incididunt quis. Cupidatat aliquip commodo fugiat occaecat nostrud ad labore mollit nostrud. Labore est ea nostrud sit reprehenderit est. Eu proident cupidatat nostrud esse dolore voluptate eu nisi. Officia et laboris tempor in mollit voluptate proident culpa proident."
    },
    {
      id: 3,
      name: "angry",
      game: "8-ball",
      details:
        "Exercitation pariatur velit id excepteur nostrud sint deserunt tempor cillum officia."
    },
    {
      id: 4,
      name: "hungry",
      game: "10-ball",
      details:
        "Minim ut minim ad cillum voluptate ad non adipisicing qui incididunt laboris. Enim incididunt voluptate cillum id adipisicing aliqua cillum quis sunt in aliquip ut commodo nisi. Ea est aliqua adipisicing sint ex id consectetur sunt ut incididunt. Deserunt irure occaecat duis cupidatat occaecat officia eiusmod sint consectetur tempor esse duis."
    },
    {
      id: 5,
      name: "poopy",
      game: "8-ball",
      details:
        "Fugiat consectetur culpa deserunt ut proident culpa labore aute. In excepteur culpa cillum magna officia do Lorem irure consequat dolore ullamco incididunt nostrud. Fugiat cupidatat quis anim anim enim consectetur ex cillum ipsum adipisicing aliquip sunt fugiat. Mollit nulla pariatur nisi minim consequat ut incididunt do incididunt voluptate id sunt. Nulla consequat minim fugiat dolore cillum velit. In qui est non do deserunt laboris dolore adipisicing occaecat. Ad officia Lorem fugiat laborum proident."
    },
    {
      id: 6,
      name: "irritated",
      game: "9-ball",
      details:
        "Qui velit voluptate Lorem elit aliquip ipsum mollit excepteur veniam exercitation ullamco sint consequat. Cupidatat ad aute ad excepteur aliquip mollit nisi. Et in minim irure elit duis. Non incididunt enim consectetur laboris et."
    },
    {
      id: 7,
      name: "flexible",
      game: "9-ball",
      details:
        "Id nulla qui sint reprehenderit eu enim aliqua. Eiusmod tempor occaecat adipisicing exercitation dolor consectetur non ipsum nostrud. Consequat consectetur proident velit sit dolore. Velit eu dolore minim commodo labore commodo reprehenderit velit dolore laborum est mollit. Quis sunt occaecat non sint dolore elit. Ullamco magna deserunt ad veniam aliqua exercitation dolore velit magna adipisicing. Non qui eu occaecat eu officia elit magna Lorem amet ex veniam proident elit ad."
    },
    {
      id: 8,
      name: "meowish",
      game: "10-ball",
      details:
        "Eu Lorem reprehenderit dolor commodo deserunt ex nulla veniam. Ad aute amet ullamco minim veniam duis tempor sit id. Ullamco elit cupidatat dolor non ea dolore labore ea veniam dolore ea anim. Duis reprehenderit ad ut incididunt quis. Nulla consequat sunt enim consectetur incididunt ipsum. Do qui enim occaecat sint eiusmod do cillum eu ullamco Lorem tempor fugiat ipsum nisi. Duis cillum anim sit labore anim veniam duis ex proident adipisicing non aute eiusmod."
    },