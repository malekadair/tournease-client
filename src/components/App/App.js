import React, { Component } from "react";
import "./App.css";
import Nav from "../Nav/Nav";
import Footer from "../Footer/Footer";
import Home from "../Home/Home";
import CreateTournament from "../CreateTournament/CreateTournament";
import TournamentDetails from "../TournamentDetails/TournamentDetails";
import Login from "../Login/Login";
import dummyStore from "./dummyStore";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";

class App extends Component {
  constructor() {
    const { tourneys } = dummyStore;
    super();
    this.state = {
      tournaments: tourneys,
      isLoading: false
    };
  }
  handleSubmit = (event, data) => {
    event.preventDefault();
    const newTournaments = [...this.state.tournaments, data];
    console.log("newtournamanets:", newTournaments);
    this.setState({
      tournaments: newTournaments
    });
  };
  render() {
    return (
      <Router>
        <div className="app">
          <Nav />
          <Switch>
            <Route
              exact
              path="/"
              render={routeProps => {
                const { tournaments, isLoading } = this.state;
                return (
                  <Home
                    {...routeProps}
                    tournaments={tournaments}
                    isLoading={isLoading}
                  />
                );
              }}
            />
            <Route
              exact
              path="/create"
              render={routeProps => {
                return (
                  <CreateTournament
                    tournamentsLength={this.state.tournaments.length + 1}
                    handleSubmit={this.handleSubmit}
                    {...routeProps}
                  />
                );
              }}
            />
            <Route
              exact
              path="/login"
              render={routeProps => {
                return (
                  <Login
                    tournamentsLength={this.state.tournaments.length + 1}
                    handleSubmit={this.handleSubmit}
                    {...routeProps}
                  />
                );
              }}
            />
            <Route
              path="/:tournid"
              render={routeProps => {
                const { tournaments, isLoading } = this.state;
                const { tournid } = routeProps.match.params;
                const foundTournament = tournaments.find(
                  ({ id }) => id == tournid
                );

                return (
                  <TournamentDetails
                    {...routeProps}
                    tournament={foundTournament}
                    isLoading={isLoading}
                  />
                );
              }}
            />
          </Switch>
          <Footer />
        </div>
      </Router>
    );
  }
}

export default App;
