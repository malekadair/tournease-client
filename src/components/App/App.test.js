import React from "react";
import toJson from "enzyme-to-json";
import App from "./App";

describe(`App component`, () => {
  const props = {
    id: "a",
    name: "test-class-name"
  };

  it("renders a .App by default", () => {
    const wrapper = <App />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the App given props", () => {
    const wrapper = <App {...props} />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
