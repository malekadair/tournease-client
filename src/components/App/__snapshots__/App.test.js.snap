// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders a .App by default 1`] = `undefined`;

exports[`App component renders the App given props 1`] = `undefined`;
